doc/user-guide.md

```diff
@@ -1 +1,384 @@
+- working directory -- the tree of all open sub-repos
+- index -- the combined indices of all open sub-repos
+
+Some implications:
+
+- Unless explicitly specified (and then only very rarely), "committishes"
+  (reference names, SHAs, etc.) provided to git-meta commands are resolved only
+  in the meta-repo.
+- The HEADs of submodules are adjusted only as a consequence of changing the
+  HEAD of the meta-repo.
+- Paths in git-meta commands are allowed to refer to any part of the working
+  tree, regardless of current working directory.
+- The working tree and index of the meta-repo itself are ignored unless the
+  user explicitly needs to add files to it. 
+- New commits in submodules (not yet reflected in the HEAD of the meta-repo)
+  are considered to be staged in the logical index of the mono-repo (and show
+  that way in `git meta status`).
+- Submodules in submodules are not currently supported, but will be treated by
+  git-meta as "normal" Git submodules, not as part of the mono-repo --
+  recursive sub-repos would add extra complication to the implementation and
+  user model, and are not required to meet the goal of creating a mono-repo.
+
+### Value-added behavior
+
+To support this model, git-meta commands offer the following conveniences not
+supported by normal Git commands:
+
+- the ability to operate on the entire mono-repo when the current working
+  directory is within a submodule, e.g.:
+
+```bash
+$ git meta open my-sub-repo
+$ cd my-sub-repo/foo/bar
+$ git checkout -b a-new-branch       # makes a branch in the sub-repo
+$ git meta checkout -b a-new-branch  # makes a branch in the meta-repo
+```
+
+- the ability to target paths within any open submodule regardless of current
+  working directory, e.g.:
+
+```bash
+$ echo >> sub-repo-x/README.md
+$ cd sub-repo-y
+$ git meta add ../sub-repo-x/README.md
+```
+
+- the ability to target multiple sub-repos in one command:
+
+```bash
+$ git meta commit sub-repo-x/README.md sub-repo-y/foo/bar/main.cpp
+```
+
+or even by directory tree:
+
+```bash
+$ echo "a change" >> my-product/sub-repo-a/README.md
+$ echo "another change" >> my-product/sub-repo-b/README.md
+$ git meta add my-product
+```
+
+- removes the need to explicitly stage and commit updated submodules:
+
+Normally in Git, if you add a commit to a submodule you need to stage the
+submodule too:
+
+```bash
+$ cd my-sub-repo
+$ echo >> foo
+$ git commit -m "changed foo" foo
+$ cd ..
+$ git commit -m "changed my-sub-repo"
+On branch master
+Changes not staged for commit:
+        modified:   my-sub-repo (new commits)
+
+no changes added to commit
+```
+
+This process is greatly simplified with git-meta:
+
+```bash
+$ echo >> my-sub-repo/foo
+$ git meta-commit -m "changed foo" my-sub-repo/foo
+```
+
+Even if you want to explicitly make two commits, you do not need to manually
+stage the sub-repo:
+
+```bash
+$ cd my-sub-repo
+$ echo >> foo
+$ git commit -m "changed foo" foo
+$ git meta commit -m "changed my-sub-repo"
```

Comment:
`git -C commit -a` differs in two ways:

- it would automatically stage and commit modified files in the meta-repo
- it would not create commits in submodules that have staged files

In the near future, the meta version will support <path> arguments, which will be submodule-aware in the same way that `git meta add` is.

What about the command strikes you as counter intuitive and/or would disrupt your workflows?